src/app/USERS/login/login.component.html

```diff
@@ -0,0 +1,18 @@
+
+    <h2>Login</h2>
+    <form [formGroup]="loginForm" (ngSubmit)="login()">
+      <div class="form-group">
+        <label for="username">Username:</label>
+        <input id="username" formControlName="username" type="text" required />
+      </div>
+  
+      <div class="form-group">
+        <label for="password">Password:</label>
+        <input id="password" formControlName="password" type="password" required />
+      </div>
+  
+      <div *ngIf="errorMessage" class="error-message">{{ errorMessage }}</div>
```

Comment:
instead of using errorMessage you can use loginForm.invalid and display a custom message